packages/bff/src/utilities/fileutils.ts

```diff
@@ -0,0 +1,215 @@
+  fileDocumentRegistry: FileDocumentsRegistry,
+  eservice: catalogApi.EService,
+  descriptor: catalogApi.EServiceDescriptor
+) {
+  return {
+    name: eservice.name,
+    description: eservice.description,
+    technology: eservice.technology,
+    mode: eservice.mode,
+    descriptor: {
+      interface: descriptor.interface && {
+        prettyName: descriptor.interface.prettyName,
+        path: descriptor.interface.path,
+      },
+      docs: descriptor.docs.map((doc) => {
+        const uniqueName = getUniqueNameByDocumentId(
+          fileDocumentRegistry,
+          doc.id
+        );
+        return {
+          prettyname: doc.prettyName,
+          path: `documents/${uniqueName}`,
+        };
+      }),
+      audience: descriptor.audience,
+      voucherLifespan: descriptor.voucherLifespan,
+      dailyCallsPerConsumer: descriptor.dailyCallsPerConsumer,
+      dailyCallsTotal: descriptor.dailyCallsTotal,
+      description: descriptor.description,
+      agreementApprovalPolicy: descriptor.agreementApprovalPolicy,
+      riskAnalysis: eservice.riskAnalysis.map((ra) => ({
+        name: ra.name,
+        riskAnalysisForm: {
+          version: ra.riskAnalysisForm.version,
+          singleAnswers: ra.riskAnalysisForm.singleAnswers.map((sa) => ({
+            key: sa.key,
+            value: sa.value,
+          })),
+          multiAnswers: ra.riskAnalysisForm.multiAnswers.map((ma) => ({
+            key: ma.key,
+            values: ma.values,
+          })),
+        },
+      })),
+    },
+  };
+}
+
+async function readableToInt8Array(readable: Readable): Promise<Uint8Array> {
+  // Collect the data from the stream
+  const chunks = [];
+  for await (const chunk of readable) {
+    chunks.push(chunk);
+  }
+
+  // Combine all chunks into a single buffer
+  const buffer = Buffer.concat(chunks);
+
+  // Create and return an Int8Array from the buffer
+  return new Uint8Array(buffer);
+}
+
+/* 
+  This function creates a zip file fetched from the S3 bucket 
+  using FileManager, the zip file containing the following files:
+  - descriptor's interface file
+  - descriptor's documents file
+  - configuration file
+
+  The zip folder structure in output is the following:
+  - zipFolderName
+      |
+      |- interface
+      |     |
+      |     |- interfaceFile.{fileExtension}
+      |
+      |- documents
+      |     |
+      |     |- documentFile1.{fileExtension}
+      |     |- documentFile2.{fileExtension}
+      |     |- ...
+      |
+      |- configuration.json
+*/
+export async function createdescriptorDocumentZipFile(
+  s3BucketName: string,
+  fileManager: FileManager,
+  logger: Logger,
+  zipFolderName: string,
+  eservice: catalogApi.EService,
+  descriptorId: string
+): Promise<Uint8Array> {
+  const descriptor = retrieveEserviceDescriptor(eservice, descriptorId);
+  const interfaceDocument = descriptor.interface;
+  if (!interfaceDocument) {
+    throw missingInterface(eservice.id, descriptorId);
+  }
+
```

Comment:
here I think that we miss the old `verifyExportEligibility` check

Reply:
Good catch, thank you! this [commit](https://github.com/pagopa/interop-be-monorepo/pull/882/commits/d9ded6f5376278a673ce770557918f9f9015f77c) adds missed validation